test/differentiability/timestepping.jl

```diff
@@ -70,5 +70,18 @@
         fd_vjp = FiniteDifferences.j′vp(central_fdm(11,1), timestep_wrt_gravity, dprogn_2, model.planet.gravity)
 
         @test isapprox(dmodel.planet.gravity, fd_vjp[1],rtol=1e-1)
+
+        # test that we can differentiate wrt to the IC while keeping model activity const 
+
+        d_progn = zero(progn)
+        d_diag = make_zero(diagn)
+
+        progn_new = zero(progn)
+        dprogn_new = one(progn) # seed 
+    
+        autodiff(set_runtime_activity(Reverse), timestep_oop!, Const, Duplicated(progn_new, dprogn_new), Duplicated(progn_copy_2, d_progn), Duplicated(diagn_copy_2, d_diag), Const(dt), Const(model))
```

Comment:
The aim of this PR is basically that his here is possible. That we denote `model` with `Const` activity. 

Ideally, this would also work without turing on the `set_runtime_activity` of Enzyme, but currently it doesn't for reasons I don't fully understand. That being said, this runtime activity mode just gives a minor speed penalty.

Reply:
`Const` isn't actually constant it just has to be constant wrt to the differentiation right?